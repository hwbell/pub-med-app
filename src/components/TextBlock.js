import React from 'react';
import '../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button } from 'reactstrap';
import { Link } from 'react-router-dom';
const Router = require("react-router-dom").BrowserRouter;


class TextBlock extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      results: null
    };
  }

  render() {
    // console.log(this.props.linkTo)

    return (
      <div className={this.props.class || "glass"} style={styles.content}>
        <p className="thread-title">{this.props.text.title}</p>

        <div>
          {this.props.text.paragraph.map((el, i) =>
            <p key={i} className="thread-text" style={{ alignSelf: 'flex-start' }}>{el}</p>
          )}
        </div>

        <div style={{margin: '0px 14px'}}>
          {this.props.buttons.map((button, i) =>
            <Button key={i} color='primary' size="sm" className="article-button">
              <Link to={button.link || '/about'} style={{ color: 'white' }}>{button.text}</Link>
            </Button>
          )}
        </div>

      </div>

    );
  }
}

const styles = {
  content: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start'
  },
}

export default TextBlock;
