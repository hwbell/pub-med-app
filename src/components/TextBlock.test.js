import React from 'react';
import ReactDOM from 'react-dom';
import TextBlock from './TextBlock';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// some mock data. test with data that has different lengths for the paragraph array
const aboutIntro = {
  title: 'About PMC',
  paragraph: [
    `PubMed Central (PMC) is a free digital repository that archives 
  publicly accessible full-text scholarly articles that have been published within the 
  biomedical and life sciences journal literature. As one of the major research 
  databases within the suite of resources that have been developed by the National 
  Center for Biotechnology Information (NCBI), PubMed Central is much more than just 
  a document repository. Launched in February 2000, the repository has grown rapidly 
  as the NIH Public Access Policy is designed to make all research funded by the 
  National Institutes of Health (NIH) freely accessible to anyone, and, in addition, 
  many publishers are working cooperatively with the NIH to provide free access to 
  their works.`,
  ],
  button: 'read more'
}

const resourcesIntro = {
  title: 'Resources',
  paragraph: [
    'abstracts > 34 million',
    'full text articles > 5 million',
    'patents > 4 million',
    'preprints > 74,000',
    'agricola records > 700,000',
    'NIH clinical guidelines > 800'
  ],
  button: 'find a resource'
}

it('renders without crashing', async () => {
  let wrapper = mount(<TextBlock text={aboutIntro}/>);
});

it('renders correctly', async () => {
  const tree = await renderer
    .create(<TextBlock text={aboutIntro}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the correct elements for different data', () => {
  let wrapper = mount( <TextBlock text={aboutIntro}/>);
  
  expect(wrapper.html()).toContain('div class=\"glass\"');
  expect(wrapper.html()).toContain('h4 class=\"subtitle\"');
  
  expect(wrapper.find('.subtitle').length).toEqual(1);
  expect(wrapper.find('Button').length).toEqual(1);
  expect(wrapper.find('.text').length).toEqual(1);

  wrapper = mount( <TextBlock text={resourcesIntro}/>);
  
  expect(wrapper.html()).toContain('div class=\"glass\"');
  expect(wrapper.html()).toContain('h4 class=\"subtitle\"');
  
  expect(wrapper.find('.subtitle').length).toEqual(1);
  expect(wrapper.find('Button').length).toEqual(1);
  expect(wrapper.find('.text').length).toEqual(6);

})