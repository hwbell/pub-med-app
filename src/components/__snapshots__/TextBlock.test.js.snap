// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="glass"
  style={
    Object {
      "height": "100%",
    }
  }
>
  <h4
    className="subtitle"
    style={
      Object {
        "alignSelf": "flex-start",
        "margin": "10px",
      }
    }
  >
    About PMC
  </h4>
  <p
    className="text"
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
  >
    PubMed Central (PMC) is a free digital repository that archives 
  publicly accessible full-text scholarly articles that have been published within the 
  biomedical and life sciences journal literature. As one of the major research 
  databases within the suite of resources that have been developed by the National 
  Center for Biotechnology Information (NCBI), PubMed Central is much more than just 
  a document repository. Launched in February 2000, the repository has grown rapidly 
  as the NIH Public Access Policy is designed to make all research funded by the 
  National Institutes of Health (NIH) freely accessible to anyone, and, in addition, 
  many publishers are working cooperatively with the NIH to provide free access to 
  their works.
  </p>
  <button
    aria-label={null}
    className="block-button btn btn-secondary btn-sm"
    onClick={[Function]}
  >
    read more
  </button>
</div>
`;
